### PR TITLE
Secondary

### PR DESCRIPTION
**Pull request by csestak-solvd**
not everything is perfect. but classes set up. need to look into arraylists

### CODE REVIEW

File: .idea/airport-app.iml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
```
Comment (ankochergin) [inline .idea/airport-app.iml]: add .idea folder and .iml files to the gitignore

File: src/Airplane.class
Comment (ankochergin) [inline src/Airplane.class:1]: do not commit .class files, only .java files

File: src/Airline.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+public class Airline {
+    private String company;
```
Comment (ankochergin) [inline src/Airline.java]:
```suggestion
public class Airline {

    private String company;
```

File: src/Airplane.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+public class Airplane implements Board, Arrive, Depart {
+    private int capacity;
```
Comment (ankochergin) [inline src/Airplane.java]:
```suggestion
public class Airplane implements Board, Arrive, Depart {

    private int capacity;
```

Comment (ankochergin) [reply src/Airplane.java]: add an empty line between the class name and first variable all files

File: src/Arrive.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+public interface Arrive {
```
Comment (ankochergin) [inline src/Arrive.java]: I was thinking about Board, Arrive, Depart interfaces. Probably we can create only 1 interface IPlane and add all plane-related methods there?

Comment (ankochergin) [reply src/Arrive.java]: Also add I letter at the beginning of the interface

File: src/FoodCourt.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+public class FoodCourt{
```
Comment (ankochergin) [inline src/FoodCourt.java]: formatting

File: src/Main.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+        Parking parking = new Parking("Extended parking", 55, 15);
+        parking.setLot("Extended Parking");
+        parking.setStall(55);
+        parking.setPrice(15);
+
+        Person person = new Person(birthDate, "Ryan Reynolds", "Male");
+//        person.setAge(35);
+//        person.setName("Ryan Reynolds");
+//        person.setSex("Male");
```
Comment (ankochergin) [inline src/Main.java]: please avoid commented lines in the code, uncomment them or remove if you don't they are not needed

File: src/airport/Airline.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package airport;
+
+public class Airline {
```
Comment (ankochergin) [inline src/airport/Airline.java:3]: I was reviewing PR from the beginning and only now noticed that all your classes are duplicated, I believe you accidentally copied them twice: to the src folder and to your new packages. please remove unneeded classes.

File: src/Main.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        FoodCourt foodCourt = new FoodCourt("Pizza", "Dominoes", 12);
+
+
+
+        Restroom restroom = new Restroom("Male", true, true);
+
+
+        Gate gate = new Gate("A123", "Boarding", "Regional");
+
+    }
```
Comment (ankochergin) [inline src/Main.java]: no empty lines at the end of a method

File: src/Main.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+
+        Airport airport = new Airport("Hobby Airport", "Houston", "International");
+
+
+        FoodCourt foodCourt = new FoodCourt("Pizza", "Dominoes", 12);
+
+
+
+        Restroom restroom = new Restroom("Male", true, true);
```
Comment (ankochergin) [inline src/Main.java]: no more than 1 empty line

File: src/airport/Airline.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+    private String company;
```
Comment (ankochergin) [inline src/airport/Airline.java]:
```suggestion
    private String companyName;
```

Comment (ankochergin) [reply src/airport/Airline.java]: after this change also change getter and setter names

File: src/airport/Airplane.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+public class Airplane implements IPlane {
+
+    private int capacity;
+    private int tailNumber;
+    private Flight flight;
+
+    public Airplane(int capacity, int tailNumber) {
+
+        this.capacity = capacity;
```
Comment (ankochergin) [inline src/airport/Airplane.java]:
```suggestion
    public Airplane(int capacity, int tailNumber) {
        this.capacity = capacity;
```

File: src/airport/Airport.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    private void setLocation() {
+        this.location = location;
+    }
+
+    private void setType() {
+        this.type = type;
+    }
+
+
+    @Override
```
Comment (ankochergin) [inline src/airport/Airport.java]:
```suggestion
    }

    @Override
```

File: src/airport/Flight.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+public class Flight {
+    
+    private String departureTime;
+    private String arrivalTime;
+    private int price;
+    private ArrayList<Passenger> passengers;
+    private Airline airline;
+
+    public Flight (String departureTime, String arrivalTime, int price){
```
Comment (ankochergin) [inline src/airport/Flight.java]:
```suggestion
    public Flight(String departureTime, String arrivalTime, int price){
```

File: src/airport/Flight.java [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+        this.departureTime = departureTime;
+        this.arrivalTime = arrivalTime;
+        this.price = price;
+    }
```
Comment (ankochergin) [inline src/airport/Flight.java:19]: do you need any getters/setters in this class?

File: src/airport/FoodCourt.java [hunk 15]
```diff
@@ -0,0 +1,39 @@
+package airport;
+
+public class FoodCourt {
+    
+    public static String foodType;
+    public static String restaurantName;
+    public static double price;
```
Comment (ankochergin) [inline src/airport/FoodCourt.java]: I think static is not the best option here

File: src/airport/Luggage.java [hunk 16]
```diff
@@ -0,0 +1,28 @@
+package airport;
+
+public class Luggage {
+    
+    int count;
+    int weight;
```
Comment (ankochergin) [inline src/airport/Luggage.java]: don't forget about access modifiers

File: src/people/Passenger.java [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+        this.security = security;
+    }
+
+
+    public boolean isAssistanceNeeded() {
+        return assistance;
+    }
+
+
+    public boolean hasGuardian() {
```
Comment (ankochergin) [inline src/people/Passenger.java]:
```suggestion
    }

    public boolean hasGuardian() {
```

File: src/airport/FoodCourt.java (hunk 15 above)
Comment (csestak-solvd) [reply src/airport/FoodCourt.java]: not sure why i would have even put static there considering everything. removed.

File: src/airport/Airport.java [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+public class Airport {
+
+    private String name;
+    private String location;
+    private String type;
+    private ArrayList<Airline> airlines;
+    private ArrayList<Employee> employees;
+    private ArrayList<Restroom> restrooms;
+    private ArrayList<FoodCourt> foodCourts;
+    private ArrayList<Parking> parkingLots;
```
Comment (ankochergin) [inline src/airport/Airport.java:15]: for these lists add methods like addEmployee(employee), addAirline(airline), addAirplane(airplane) to fill these lists

File: src/interfaces/IPlane.java [hunk 19]
```diff
@@ -0,0 +1,13 @@
+package interfaces;
+
+public interface IPlane {
+
+    String arrPlane = "Boeing 757";
```
Comment (ankochergin) [inline src/interfaces/IPlane.java:5]: I think arrPlane and depPlane are not a part of the interface, please remove them for now. Interface is just like a contract, if class implements it then class should override all methods from the interface.

File: src/people/Person.java [hunk 20]
```diff
@@ -0,0 +1,50 @@
+package people;
+import interfaces.IEat;
+import airport.FoodCourt;
+import java.time.LocalDate;
+
+public class Person implements IEat {
```
Comment (ankochergin) [inline src/people/Person.java:6]: I think it should be abstract

Comment (csestak-solvd) [reply src/people/Person.java:6]: done

File: src/interfaces/IPlane.java (hunk 19 above)
Comment (csestak-solvd) [reply src/interfaces/IPlane.java:5]: removed

File: src/airport/Airport.java (hunk 18 above)
Comment (csestak-solvd) [reply src/airport/Airport.java:15]: Completed

**Pull request merged**